src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Telemetry/ITelemetryReporter.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace Microsoft.AspNetCore.Razor.Telemetry;
 
-internal interface ITelemetryReporter
+internal interface ITelemetryReporter : IDisposable
```

Comment:
Does this interface need to be `IDisposable`? Or, is it sufficient to mark `TelemetryReporter` as `IDisposable`? AFAICT, this isn't needed by the `ITelemetryReporter` contract itself and results in a do-nothing `Dispose()` on `NoOpTelemetryReporter`.

Reply:
This is where my knowledge gets a little iffy. Disposal is handled by the DI implementation, and it's added as a singleton with `AddSingleton<ITelemetryReporter>`. Does DI know if the underlying class needs to be disposed? Do I need to add a finalizer pattern with disposal?

Reply:
Yes, all the DI containers we use -- MEF, Microsoft.Extensions.DependencyInjection`, VSMEF -- will dispose the runtime object if it implements `IDisposable`. It is not necessary for the contract interface to be `IDisposable`. (Actually, it would be pretty weird if that was a requirement!)

Reply:
I just have blissful ignorance for MEF and VSMEF :) Will move IDisposable down